### PR TITLE
fix(picker): prevent diacritic text cutoff

### PR DESCRIPTION
**Pull request by rise-erpelding**
## Description

Addressing [CSS-810](https://jira.corp.adobe.com/browse/CSS-810), this fix addresses [a bug that was called out recently](https://adobedesign.slack.com/archives/CESK60MQD/p1718744842850969), similar to what was seen in Textfield for both [Spectrum CSS](https://github.com/adobe/spectrum-css/pull/2123) and [React Spectrum](https://github.com/adobe/react-spectrum/pull/4579) with text with diacritics being cut off vertically. In this case, as the affected text occurs in a `<span>` rather than an `<input>` tag, the line-height does not need an adjustment, and changing the margin to padding appears to be sufficient to allow the vertical cutoff to be displayed.

There is also a [draft PR](https://github.com/adobe/spectrum-css/pull/2913) to fix the same issue that uses the same line-height fix that is seen in textfield.

_Note: When testing with Assistiv Labs, this cutoff issue did not appear to be present (in production)--this fix has been tested in Assistiv Labs (Chrome, Firefox, Edge) and does not appear to have an impact there._

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps
1. Check out the Picker in Storybook locally or in the [deploy preview](https://pr-2914--spectrum-css.netlify.app/preview/?path=/story/components-picker--default)
2. Change text to Thai:
    - In the Storybook controls, you can change the text for both "Label" and "Placeholder" to the Thai phrase that was causing an issue previously, "ทั้งหมด". You can also do this in the [deployed version of Picker](https://opensource.adobe.com/spectrum-css/preview/index.html?path=/story/components-picker--default) or from the `main` branch to see the cutoff issue.
    - If checking out Picker locally, you can alternatively cherry pick [commit](https://github.com/adobe/spectrum-css/pull/2914/commits/a1fab905fba48b9d3cf641a19160bee1b07e376a) `a1fab905fba48b9d3cf641a19160bee1b07e376a` to temporarily change the English in Picker to Thai. This was a temporary commit I'd put in the PR but reverted in order to run VRTs.
3. Confirm that the text within the Picker is not being cut off vertically. The best way to know is to compare the field label ("Label" text, which has not been experiencing cutoff) with the picker label ("Placeholder" text, which is the element experiencing cutoff), applying zoom as needed.
4. Confirm that this continues to be true for:
    - [x] all sizes of Picker
    - [x] all variants of Picker
    - [x] in more than one browser (I checked Chrome, Firefox, Edge, and Safari, all for Mac)
5. Check the [Tabs component in Storybook](https://github.com/adobe/spectrum-css/pull/2914/commits/a1fab905fba48b9d3cf641a19160bee1b07e376a):
    - [x] confirm that there do not appear to be any regressions or issues as the result of this change to Picker

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [x] The pages render correctly, are accessible, and are responsive.

9. If components have been modified, VRTs have been run on this branch:

- [X] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes--there is a 2px length difference in the snapshots for the Tabs component, this cannot be replicated in the browser, and I am unsure if it's an indication of significant diffs that might be visible downstream.

## Screenshots

Before:
image

After:
image

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] I have tested these changes in Windows High Contrast mode (and AssistivLabs).
- [x] If my change impacts **other components**, I have tested to make sure they don't break.
- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

**Review by marissahuysentruyt — APPROVED** [review]
This seems like a nice, simple fix! I didn't see any cutoff in the `span` text in any of the sizes, I tried to combine a bunch of the states to see if I could break it. I also tested in Chrome, Firefox, Safari and Edge on Mac, but it all looks good!

My only question now is do we drop those 2 temp commits (changing the text to Thai, and then the corresponding revert)? If we merged it right now, it would be ok, but do we have a preference for "temporary" commits?

Great catch on the spelling correction too! 🎉

**Comment by rise-erpelding** [comment]
> My only question now is do we drop those 2 temp commits (changing the text to Thai, and then the corresponding revert)? If we merged it right now, it would be ok, but do we have a preference for "temporary" commits?

My original intention was to _not_ have reverted the temporary commit for validation purposes and then to drop the commit before merging, but then I reverted it to run VRTs and ran into that unexpected diff that I wanted feedback on, so I left it that way. I think I would probably just squash/fixup those commits. Also was thinking there wasn't a lot of reason to keep the spelling change for "country" as a separate commit, so that'd be squashed in with the fix here.

File: components/picker/stories/picker.stories.js [hunk 1]
```diff
@@ -207,7 +207,7 @@ const Variants = (args) => html`
 					...args,
 					isOpen: false,
 					withSwitch: true,
-					placeholder: "Select your contry of origin",
+					placeholder: "Select your country of origin",
```
Comment (pfulton) [inline components/picker/stories/picker.stories.js:210]: Thanks for catching this!

**Review by pfulton — APPROVED** [review]
Thanks for digging into this one and I'm glad it was a pretty simple fix.

**Comment by snowystinger** [comment]
Checking if there is any action I need to take in RSP.
Where was this reported against React Spectrum? it looks like we've already implemented this? https://react-spectrum.adobe.com/react-spectrum/Picker.html
Does anyone have the actual string which is cutoff? I have no idea what those characters are in the screenshot, can't copy and paste them :-/
